tests/test_api.py

```diff
@@ -526,52 +527,40 @@ def test_metadata_targets(self):
         targets_obj = Targets.from_dict(tmp_dict)
         self.assertEqual(targets_dict["signed"], targets_obj.to_dict())
 
-    def setup_dict_with_unrecognized_field(self, file_path, field, value):
-        json_dict = {}
-        with open(file_path) as f:
-            json_dict = json.loads(f.read())
-        # We are changing the json dict without changing the signature.
-        # This could be a problem if we want to do verification on this dict.
-        json_dict["signed"][field] = value
-        return json_dict
+    # Insert item into every dictionary in signed (where allowed by spec)
+    @staticmethod
+    def poison_signed(signed: Dict[str, Any], key: str):
+        named_dicts = [("signed", signed)]
+        while named_dicts:
+            name, container = named_dicts.pop()
+
+            # Process all child dicts. spec does not contain any
+            # dicts inside arrays so no need to check those
+            for valname, value in container.items():
+                if isinstance(value, dict):
+                    named_dicts.append((valname, value))
+
+            # if dicts keys are limited in spec or if dicts values are
+            # strictly defined, skip it
+            if name in ["hashes", "keys", "meta", "roles", "targets"]:
```

Comment:
Not really a review but how did we come up with this list? I guess you did by going through the spec. Can we note it somewhere so that we don't do it again?

Reply:
Yes you guessed correctly. I'll try to phrase that somehow clearly

Reply:
Wondering why do you include `meta` here?

Reply:
because it seems to me the meta content is strictly defined: keys must be metadata filenames. values must be MetaFile dict representations. I don't think allowing something else is reasonable.